okhttp/src/main/java/com/squareup/okhttp/internal/spdy/SpdyStream.java

```diff
@@ -218,6 +231,12 @@ public void close(ErrorCode rstStatusCode) throws IOException {
     if (!closeInternal(rstStatusCode)) {
       return; // Already closed.
     }
+    // Close down dependant pushed streams
+    for(SpdyStream pushed : pushedStreams) {
+      int pushedId = pushed.getId();
+      pushed.getSource().close();
+      pushedStreams.remove(pushedId);
```

Comment:
pushedId is an integer, this will result in pushedStreams never getting cleared. Also, this whole business of closing other streams is fishy - I'm going to see what happens if we remove all of this associated stream and pushed stream code changes and see what happens.